### PR TITLE
*: Fix a bug in type inferrer

### PR DESCRIPTION
**Pull request by shenli**
Missing CaseExpr
Pass https://travis-ci.org/qiuyesuifeng/sqllogictest-groupby

@qiuyesuifeng @coocood

### CODE REVIEW

File: optimizer/typeinferer.go [hunk 1]
```diff
@@ -275,3 +284,43 @@ func (v *typeInferrer) handleFuncCallExpr(x *ast.FuncCallExpr) {
...
+		if c < 0 {
+			currType = t
+		}
+	}
+	if x.ElseClause != nil {
+		t := x.ElseClause.GetType()
+		if currType == nil {
+			currType = t
+		}
+		c := v.compareType(currType.Tp, t.Tp)
```
Comment (qiuyesuifeng) [inline optimizer/typeinferer.go]: Should this in else condition?

**Comment by shenli** [comment]
@coocood @qiuyesuifeng

**Comment by coocood** [comment]
LGTM

**Comment by qiuyesuifeng** [comment]
LGTM

**Pull request merged**